firestore/integration_test_internal/src/bundle_test.cc

```diff
@@ -0,0 +1,298 @@
+                                             {"bar", FieldValue::Integer(2)}}));
+    }
+
+    {
+      Query limit = AwaitResult(db->NamedQuery(kLimitQueryName));
+      auto limit_snapshot = AwaitResult(limit.Get(Source::kCache));
+      EXPECT_THAT(
+          QuerySnapshotToValues(limit_snapshot),
+          testing::ElementsAre(MapFieldValue{{"k", FieldValue::String("b")},
+                                             {"bar", FieldValue::Integer(2)}}));
+    }
+
+    {
+      Query limit_to_last = AwaitResult(db->NamedQuery(kLimitToLastQueryName));
+      auto limit_to_last_snapshot =
+          AwaitResult(limit_to_last.Get(Source::kCache));
+      EXPECT_THAT(
+          QuerySnapshotToValues(limit_to_last_snapshot),
+          testing::ElementsAre(MapFieldValue{{"k", FieldValue::String("a")},
+                                             {"bar", FieldValue::Integer(1)}}));
+    }
+  }
+};
+
+TEST_F(BundleTest, CanLoadBundlesWithoutProgressUpdates) {
+  Firestore* db = TestFirestore();
+  auto bundle = CreateTestBundle(db);
+
+  Future<LoadBundleTaskProgress> result = db->LoadBundle(bundle);
+
+  VerifySuccessProgress(AwaitResult(result));
+  VerifyQueryResults(db);
+}
+
+TEST_F(BundleTest, CanLoadBundlesWithProgressUpdates) {
+  Firestore* db = TestFirestore();
+  auto bundle = CreateTestBundle(db);
+
+  std::vector<LoadBundleTaskProgress> progresses;
+  Future<LoadBundleTaskProgress> result = db->LoadBundle(
+      bundle, [&progresses](const LoadBundleTaskProgress& progress) {
+        progresses.push_back(progress);
+      });
+
+  auto final_progress = AwaitResult(result);
+
+  // 4 progresses will be reported: initial, document 1, document 2, final
+  // success.
+  ASSERT_EQ(progresses.size(), 4);
+  EXPECT_THAT(progresses[0], InProgressWithLoadedDocuments(0));
+  EXPECT_THAT(progresses[1], InProgressWithLoadedDocuments(1));
+  EXPECT_THAT(progresses[2], InProgressWithLoadedDocuments(2));
+  VerifySuccessProgress(progresses[3]);
+  EXPECT_EQ(progresses[3], final_progress);
+
+  VerifyQueryResults(db);
+}
+
+TEST_F(BundleTest, CanDeleteFirestoreFromProgressUpdate) {
+  Firestore* db = TestFirestore();
+  auto bundle = CreateTestBundle(db);
+
+  std::promise<void> db_deleted;
+  std::vector<LoadBundleTaskProgress> progresses;
+  Future<LoadBundleTaskProgress> result =
+      db->LoadBundle(bundle, [this, db, &progresses, &db_deleted](
+                                 const LoadBundleTaskProgress& progress) {
+        progresses.push_back(progress);
+        // Delete firestore before the final progress.
+        if (progresses.size() == 3) {
+          DeleteFirestore(db);
+          db_deleted.set_value();
+        }
+      });
+
+  // Wait for the notification that the instance is deleted. We cannot rely on
+  // the returned future because it will not complete.
+  db_deleted.get_future().wait();
+
+  // This future is not completed, and returns back a nullptr for result.
+  EXPECT_EQ(Await(result), nullptr);
```

Comment:
The comment a few lines above at line 167 says "We cannot rely on the returned future because it will not complete" and on line 171 says "This future is not completed". If the `Future` is known to not complete, then it does not make sense to await its completion. Are those comment incorrect?